[CMake] Imported targets improperly forwarded

It seems that the exported targets in ROOT 6.26 changed. Some software ([corryvreckan](https://gitlab.cern.ch/corryvreckan/corryvreckan)) that was compiling without issues in older root versions encounters issues when running cmake (3.22.3). 

I believe that this is related to missing exported CMake targets.

### Describe the bug

    CMake Error at src/core/CMakeLists.txt:73 (ADD_LIBRARY):
      Target "CorryvreckanCore" links to target "Threads::Threads" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    
    
    CMake Error at src/core/CMakeLists.txt:73 (ADD_LIBRARY):
      Target "CorryvreckanCore" links to target "OpenMP::OpenMP_CXX" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at src/core/CMakeLists.txt:73 (ADD_LIBRARY):
      Target "CorryvreckanCore" links to target "MPI::MPI_CXX" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?

The dependency seems to be pulled in via Minuit2. 

The following additions to the CMakeLists of the project fix the issue:

```
if(NOT TARGET Threads::Threads)
  find_package(Threads)
endif()
if(NOT TARGET MPI::MPI_CXX)
  find_package(MPI)
endif()
if(NOT TARGET OpenMP::OpenMP_CXX)
  find_package(OpenMP)
endif()
```

### Setup

Compiling from source. Tested in master and using 6.26/00. Using archlinux, using CMake 3.22.3.

The issue seems to only be present on archlinux, probably because for other OSs, the dependencies are handled internally rather than being imported (?).

--- a/cmake/modules/RootBuildOptions.cmake
+++ b/cmake/modules/RootBuildOptions.cmake
@@ -146,6 +146,8 @@ ROOT_BUILD_OPTION(mathmore ON "Build libMathMore extended math library (requires
 ROOT_BUILD_OPTION(memory_termination OFF "Free internal ROOT memory before process termination (experimental, used for leak checking)")
 ROOT_BUILD_OPTION(mlp ON "Enable support for TMultilayerPerceptron classes' federation")
 ROOT_BUILD_OPTION(minuit2 ON "Build Minuit2 minimization library")
+ROOT_BUILD_OPTION(minuit2_mpi OFF "Enable support for MPI in Minuit2")
+ROOT_BUILD_OPTION(minuit2_omp OFF "Enable support for OpenMP in Minuit2")
 ROOT_BUILD_OPTION(monalisa OFF "Enable support for monitoring with Monalisa (requires libapmoncpp)")
 ROOT_BUILD_OPTION(mpi OFF "Enable support for Message Passing Interface (MPI)")
 ROOT_BUILD_OPTION(mysql ON "Enable support for MySQL databases")

--- a/cmake/scripts/ROOTConfig.cmake.in
+++ b/cmake/scripts/ROOTConfig.cmake.in
@@ -90,6 +90,15 @@ if (NOT ROOT_builtin_nlohmannjson_FOUND)
   include(CMakeFindDependencyMacro)
   find_dependency(nlohmann_json @nlohmann_json_VERSION@)
 endif()
+if(ROOT_minuit2_mpi_FOUND)
+  include(CMakeFindDependencyMacro)
+  find_dependency(MPI)
+endif()
+if(ROOT_minuit2_omp_FOUND)
+  include(CMakeFindDependencyMacro)
+  find_dependency(OpenMP)
+  find_dependency(Threads)
+endif()
 
 #----------------------------------------------------------------------------
 # Now set them to ROOT_LIBRARIES

--- a/cmake/scripts/ROOTConfig.cmake.in
+++ b/cmake/scripts/ROOTConfig.cmake.in
@@ -86,16 +86,14 @@ endforeach()
 
 #----------------------------------------------------------------------------
 # Find external packages which were used in ROOT
+include(CMakeFindDependencyMacro)
 if (NOT ROOT_builtin_nlohmannjson_FOUND)
-  include(CMakeFindDependencyMacro)
   find_dependency(nlohmann_json @nlohmann_json_VERSION@)
 endif()
 if(ROOT_minuit2_mpi_FOUND)
-  include(CMakeFindDependencyMacro)
   find_dependency(MPI)
 endif()
 if(ROOT_minuit2_omp_FOUND)
-  include(CMakeFindDependencyMacro)
   find_dependency(OpenMP)
   find_dependency(Threads)
 endif()

--- a/math/minuit2/CMakeLists.txt
+++ b/math/minuit2/CMakeLists.txt
@@ -8,10 +8,9 @@ cmake_minimum_required(VERSION 3.1)
 
 if(NOT CMAKE_PROJECT_NAME STREQUAL ROOT)
   project(Minuit2 LANGUAGES CXX)
+  option(minuit2_mpi "Enable support for MPI in Minuit2")
+  option(minuit2_omp "Enable support for OpenMP in Minuit2")
 endif(NOT CMAKE_PROJECT_NAME STREQUAL ROOT)
-
-option(minuit2_mpi "Enable support for MPI in Minuit2")
-option(minuit2_omp "Enable support for OpenMP in Minuit2")
 
 # This package can be built separately
 # or as part of ROOT.

